### PR TITLE
Prepare docs for the release (closes #182, closes #174, closes #189)

### LINKED ISSUE #174: Add EditorConfig configuration

### LINKED ISSUE #182: Prepare the markdown documentation for v1.0.0

### LINKED ISSUE #189: Adopt JavaScript Standard Style

### PR DESCRIPTION
**Pull request by kossnocorp**
Preparing docs & things around for the release:

- Generate markdown docs for the website
- Clean up Travis CI config
- Update LICENSE file
- Add stats scripts
- Adopt JavaScript Standard Style
- Add EditorConfig config
- Update README
- Update Contributing Guide
- Adopt "Keep a CHANGELOG"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,205 +220,126 @@ console.log(isLastDayOfMonth(date))
...
+
+date-fns doesn't support historical DST changes and uses native browser API
+to detect UTC offset. Also it can't parse [IANA](https://www.iana.org/time-zones)
+time zone identifiers. If you are interested in advanced time zone
+support, checkout [Moment Timezone](http://momentjs.com/timezone/) or
+[vote for a future request](https://github.com/date-fns/date-fns/issues/180).
+
+Unlike most popular date libraries, date-fns doesn't support I18n and
+uses English as the only language. However this is matter only
+of two functions `parse` and `format`.
```
Comment (LeshaKoss) [inline README.md]: three functions `distanceInWords`, `distanceInWordsToNow` and `format`

Comment (kossnocorp) [inline README.md]: @LeshaKoss thanks!

**Comment by LeshaKoss** [comment]
@kossnocorp do you mind if I fix typos in this branch?

**Comment by kossnocorp** [comment]
@LeshaKoss nope, work is still in progress, I already made a lot of changed on top of the current state.

**Comment by kossnocorp** [comment]
@LeshaKoss I'll ping you for review

**Comment by LeshaKoss** [comment]
@kossnocorp ok!

**Comment by kossnocorp** [comment]
@LeshaKoss please take a look. It makes sense to check each commit individually because "Adopt JavaScript Standard Style" is way too big.

**Comment by LeshaKoss** [comment]
Unrelated to changes, but is there a reason why commits I see here on github.com and commits from cloned copy are ordered differently?
screen shot 2016-04-24 at 18 27 50
screen shot 2016-04-24 at 18 27 19

File: src/parse/index.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ var parseTokenTimezoneHHMM = /^([+-])(\d{2}):?(\d{2})$/
  * var result = parse('2014-02-11T11:30:30')
  * //=> Tue Feb 11 2014 11:30:30
  */
-var parse = function(dateString) {
+function parse (dateString) {
```
Comment (LeshaKoss) [inline src/parse/index.js]: @kossnocorp should we do the same for the other functions? (`var name = function () {` → `function name () {`)

File: scripts/generate_docs.js [hunk 3]
```diff
@@ -86,7 +88,7 @@ function generateDocFromSource (fn) {
 
     stream.on('data', (chunk) => { data += chunk })
     stream.on('end', () => resolve(JSON.parse(data)))
-  })
+  }).then((doc) => acc.concat(doc))
```
Comment (LeshaKoss) [inline scripts/generate_docs.js]: @kossnocorp please review this.
I made generateDocs slower, but before this change all 147 files were opened and parsed through at the same time, making CPU go 100% and effectively making my laptop unoperational while docs are being generated. I decided that faster generation isn't worth making computer freeze. What do you think?

File: scripts/generate_docs.js [hunk 4]
```diff
@@ -1,50 +1,125 @@
...
-        (acc[doc.category] = acc[doc.category] || []).push(doc)
-        return acc
-      }, groupsTemplate)
-
-    fs.writeFileSync(path.join(process.cwd(), 'dist', 'date_fns_docs.json'), JSON.stringify(groupedDocs))
-  })
+
+    stream.on('data', (chunk) => { data += chunk })
+    stream.on('end', () => resolve(JSON.parse(data)))
+  }).then((doc) => acc.concat(doc))
```
Comment (LeshaKoss) [inline scripts/generate_docs.js]: @kossnocorp please review this.
I made generateDocs slower, but before this change all 147 files were opened and parsed through at the same time, making CPU go 100% and effectively making my laptop unoperational while docs are being generated. I decided that faster generation isn't worth making computer freeze. What do you think?
(see commit 3c63bb8)

**Comment by LeshaKoss** [comment]
@kossnocorp also we need to squash all commits in this branch together, because here at this page I see commits in the wrong order and it would be unfortunate if this mess would go into `master`

File: scripts/generate_docs.js (hunk 4 above)
Comment (kossnocorp) [inline scripts/generate_docs.js]: @LeshaKoss great catch :-D.

File: package.json [hunk 5]
```diff
@@ -25,15 +25,17 @@
...
     "karma-mocha-reporter": "^1.1.1",
-    "karma-phantomjs-launcher": "^0.2.1",
+    "karma-phantomjs-launcher": "^1.0.0",
     "karma-sauce-launcher": "^0.3.0",
     "karma-sinon": "^1.0.4",
     "karma-sourcemap-loader": "^0.3.5",
     "karma-webpack": "^1.7.0",
     "mocha": "^2.4.5",
     "node-fetch": "^1.3.3",
+    "phantomjs-prebuilt": "^2.1.7",
```
Comment (kossnocorp) [inline package.json]: Cool!

File: src/parse/index.js (hunk 2 above)
Comment (kossnocorp) [inline src/parse/index.js]: @LeshaKoss this is not necessary with Standart JavaScript, but preferable.

**Comment by kossnocorp** [comment]
@LeshaKoss maybe I need to rebase once again?

**Comment by LeshaKoss** [comment]
@kossnocorp haha 😁 [--ignore-date](http://stackoverflow.com/a/1579676) option may help

**Pull request merged**